File: Week_010/cpp/src/_657_RobotReturnToOrigin.cpp
class _657_ {
public:
    bool judgeCircle(string moves) {
        int x = 0, y = 0;
        for (auto const& c : moves) {
            switch(c) {
                case 'L':
                    x--;
                    break;
                case 'R':
                    x++;
                    break;
                case 'U':
                    y++;
                    break;
                case 'D':
                    y--;
                    break;
            }
        }
        return x == 0 && y == 0;
    }
};

File: Week_010/kotlin/src/_657_RobotReturnToOrigin.kt
class _657_ {
    fun judgeCircle(moves: String): Boolean {
        var x = 0
        var y = 0
        moves.forEach {
            when (it) {
                'R' -> x++
                'L' -> x--
                'U' -> y++
                'D' -> y--
            }
        }
        return x == 0 && y == 0
    }
}